frontend/src/pages/BlogDetail/BlogDetail.js

```diff
@@ -11,7 +11,7 @@ const BlogDetail = () => {
   let navigate = useNavigate();
 
   useEffect(() => {
-    API.get(`/posts/${postSlug}/`)
+    API.get(`/postdetails/${postSlug}/`)
```

Comment:
keep it like below. The slug and the method type will determine that it is for get post details.
```
/posts/slug
//or
/post/slug
```

Then in your PostView class, create a function that will determine if it is a post detail